src/phoenix/server/api/routers/v1/dataset_examples.py

```diff
@@ -64,73 +64,83 @@ async def list_dataset_examples(request: Request) -> Response:
         )
 
     async with request.app.state.db() as session:
-        # Subquery to find the maximum created_at for each dataset_example_id
-        # timestamp tiebreaks are resolved by the largest id
-        partial_subquery = select(
-            func.max(DatasetExampleRevision.id).label("max_id"),
-        ).group_by(DatasetExampleRevision.dataset_example_id)
-
-        # if a version_id is provided, filter the subquery to only include revisions from that
-        # version or earlier
-        if version_id is not None:
-            # ensure that the specified version exists
-            matched_version_id = (
-                select(DatasetVersion.id)
-                .where(DatasetVersion.id == int(version_id.node_id))
-                .scalar_subquery()
+        if (
+            resolved_dataset_id := await session.scalar(
+                select(Dataset.id).where(Dataset.id == int(dataset_id.node_id))
             )
-            partial_subquery = partial_subquery.filter(
-                DatasetExampleRevision.dataset_version_id <= matched_version_id
+        ) is None:
+            return Response(
+                content=f"No dataset with id {dataset_id} can be found.",
+                status_code=HTTP_404_NOT_FOUND,
             )
+        if version_id:
+            if (
+                resolved_version_id := await session.scalar(
+                    select(DatasetVersion.id).where(
+                        and_(
+                            DatasetVersion.dataset_id == resolved_dataset_id,
+                            DatasetVersion.id == int(version_id.node_id),
+                        )
+                    )
+                )
+            ) is None:
+                return Response(
+                    content=f"No dataset version with id {version_id} can be found.",
```

Comment:
seems like we are accommodating an edge case, since in most scenarios this extra step would be less efficient